ui/app/AppLayouts/Communities/popups/InDropdown.qml

```diff
@@ -30,23 +38,97 @@ StatusDropdown {
         Add, Update
     }
 
-    width: 289
-    padding: 8
-
-    // force keeping within the bounds of the enclosing window
-    margins: 0
-
     signal addChannelClicked
     signal communitySelected
     signal channelsSelected(var channels)
 
     function setSelectedChannels(channels) {
-        d.setSelectedChannels(channels)
+        d.selectedChannels.clear()
+        channels.forEach(c => d.selectedChannels.add(c))
+        d.selectedChannelsChanged()
+    }
+
+    onAboutToHide: {
+        searcher.text = ""
+        listView.positionViewAtBeginning()
+    }
+
+    onAboutToShow: listView.Layout.preferredHeight = Math.min(
+                       listView.implicitHeight, 420)
+
+    // only channels (no entries representing categories), sorted according to
+    // category position and position
+    SortFilterProxyModel {
+        id: onlyChannelsModel
+
+        sourceModel: root.model
+
+        filters: ValueFilter {
+            roleName: "isCategory"
+            value: false
+        }
+
+        sorters: [
+            RoleSorter {
+                roleName: "categoryPosition"
+                priority: 2 // Higher number -> higher priority
+            },
+            RoleSorter {
+                roleName: "position"
+                priority: 1
+            }
+        ]
     }
 
-    onAboutToHide: searcher.text = ""
-    onAboutToShow: scrollView.Layout.preferredHeight = Math.min(
-                       scrollView.implicitHeight, 420)
+    // only items representing categories
+    SortFilterProxyModel {
+        id: categoriesModel
+
+        sourceModel: root.model
+
+        filters: ValueFilter {
+            roleName: "isCategory"
+            value: true
+        }
+    }
+
+    // categories, name role renamed to categoryName
+    RolesRenamingModel {
```

Comment:
This could be done in the above `categoriesModel` with the `JoinRole`, effectively renaming the role

Reply:
The goal here is to rename and therefore avoid conflict when joining in LeftJoinModel. `JoinRole` computes new role but leaves the source as it was. `RolesRenamingModel` itself is lightweight, it doesn't hurt in terms of performance, (probably) lighter than using `JoinRole`.

Reply:
Fair enough